Kickstarter-iOS/DataSources/PledgeDataSource.swift

```diff
@@ -10,26 +10,80 @@ final class PledgeDataSource: ValueCellDataSource {
     case summary
   }
 
-  func load(amount: Double, currency: String, delivery: String, isLoggedIn: Bool) {
+  enum PledgeInputRow {
+    case pledgeAmount(amount: Double, currency: String)
+    case shippingLocation(location: String, amount: Double, currencyCode: String)
+
+    var isShipping: Bool {
+      switch self {
+      case .shippingLocation: return true
+      default: return false
+      }
+    }
+
+    var isPledgeAmount: Bool {
+      switch self {
+      case .pledgeAmount: return true
+      default: return false
+      }
+    }
+  }
+
+  func load(data: PledgeTableViewData) {
+    self.loadProjectSection(delivery: data.delivery)
+
+    self.loadInputsSection(amount: data.amount, currency: data.currency, currencyCode: data.currencyCode,
+                           requiresShippingRules: data.requiresShippingRules)
+
+    self.loadSummarySection(isLoggedIn: data.isLoggedIn)
+  }
+
+  func loadSelectedShippingRule(data: SelectedShippingRuleData) {
+    guard self.numberOfItems(in: PledgeDataSource.Section.inputs.rawValue) > 1 else { return }
+
+    self.set(value: PledgeInputRow.shippingLocation(location: data.location, amount: data.amount,
+                                                    currencyCode: data.currencyCode),
+             cellClass: PledgeShippingLocationCell.self,
+             inSection: Section.inputs.rawValue,
+             row: 1)
+  }
+
+  func shippingCellIndexPath() -> IndexPath? {
+    let inputsRowCount = self.numberOfItems(in: PledgeDataSource.Section.inputs.rawValue)
```

Comment:
Not as part of this PR, but would be nice if we could do something like `self.indexPath(where: ...)` and give it a condition to match on. Feels like us keeping track of the position of things can become problematic. Although I guess it's locked in with tests 🤷‍♂